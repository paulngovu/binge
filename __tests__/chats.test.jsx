import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Chats from '../pages/chats';

configure({ adapter: new Adapter() });

describe('Chats page', () => {
  const emptyChats = {
    allMessages: {},
    foodChats: [],
    foodData: [],
    username: "testuser"
  };

  const testChats = {
    allMessages: {},
    foodChats: ['foodA'],
    foodData: [],
    username: "testuser"
  };

  it('renders chats page with empty sidebar', () => {
    const tree = renderer.create(<Chats 
        allMessages={emptyChats["allMessages"]}
        foodChats={emptyChats["foodChats"]}
        foodData={emptyChats["foodData"]}
        username={emptyChats["username"]}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders prompt message when sidebar is empty', () => {
    const chats = shallow(<Chats 
        allMessages={emptyChats["allMessages"]}
        foodChats={emptyChats["foodChats"]}
        foodData={emptyChats["foodData"]}
        username={emptyChats["username"]}
    />);
    expect(chats.contains('Chats will show up after you like a food item!')).toBeTruthy();
  });

  it('renders food name in sidebar when liked', () => {
    const chats = shallow(<Chats 
        allMessages={testChats["allMessages"]}
        foodChats={testChats["foodChats"]}
        foodData={testChats["foodData"]}
        username={testChats["username"]}
    />);
    expect(chats.contains('foodA')).toBeTruthy();
  });

});
