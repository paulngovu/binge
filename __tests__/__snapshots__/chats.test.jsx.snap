// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chats page renders chats page with empty sidebar 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 hjInHh"
>
  <header
    className="StyledBox-sc-13pk1d4-0 gpGtYH"
    display="flex"
  >
    <h1
      className="StyledHeading-sc-1rdh4aw-0 elclaf"
    >
      Binge
    </h1>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 itCgil"
    />
    <div
      alignself="end"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 wrGSV"
      >
        <span
          className="StyledText-sc-1sadyjn-0 deIuKd"
        >
          Hi, 
          testuser
        </span>
        <a
          className="StyledButton-sc-323bzc-0 haWZSW"
          data-testid="home-btn"
          href="/"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <svg
            aria-label="Home"
            className="StyledIcon-sc-ofa7kd-0 bwvewX"
            viewBox="0 0 24 24"
          >
            <path
              d="m1 11 11-9 11 9m-8 12v-8H9v8m-5 0V9m16 14V9"
              fill="none"
              stroke="#000"
              strokeWidth="2"
            />
          </svg>
        </a>
      </div>
    </div>
  </header>
  <div
    className="StyledGrid-sc-1wofa1l-0 heOMlz"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 bwlxpi"
    />
    <div
      className="StyledBox-sc-13pk1d4-0 jCDZxg"
    >
      <div />
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 gYejVn"
    >
      <span
        aria-label="default-text"
        className="StyledText-sc-1sadyjn-0 hQnoUE"
      >
        Chats will show up after you like a food item!
      </span>
    </div>
  </div>
</div>
`;
